parquet-avro/src/main/java/org/apache/parquet/avro/AvroVariantConverter.java

```diff
@@ -77,9 +84,9 @@ public void end() {
 
     builder.appendNullIfEmpty();
 
-    Object record = model.newRecord(null, avroSchema);
-    model.setField(record, "metadata", metadataPos, metadata.getEncodedBuffer());
-    model.setField(record, "value", valuePos, builder.encodedValue());
+    Object record = model.newRecord(null, VARIANT_SCHEMA);
```

Comment:
I think we need this change that we will produce a fixed schema of value and metadata rather than reading the original avro schema.

Also, we also fix the issue that value field may be missing from the schema since that is allowed, i.e., typed_value exists but value doesn't. We don't read value field for the position and the output schema should be fixed.

Reply:
We can't change the expected Avro schema. We can reject it, but the contract is to use the schema that was passed in. I think the earlier code is correct.

Reply:
Here what we are passing in the shredded schema which may/may not have `value` or `typed_value` field. 

But we want to generate a variant which maps to a record of `value` and `metadata` fields, right?

<img width="693" height="136" alt="image" src="https://github.com/user-attachments/assets/007b8b31-ea8b-4c47-acc5-57533a1a3d99" />